src/coreclr/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -55,13 +55,32 @@ private IDictionary CreateDataContainer()
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern bool IsImmutableAgileException(Exception e);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern IRuntimeMethodInfo GetMethodFromStackTrace(object stackTrace);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ExceptionNative_GetMethodFromStackTrace")]
+        private static partial void GetMethodFromStackTrace(ObjectHandleOnStack stackTrace, ObjectHandleOnStack method);
 
         private MethodBase? GetExceptionMethodFromStackTrace()
         {
             Debug.Assert(_stackTrace != null, "_stackTrace shouldn't be null when this method is called");
-            IRuntimeMethodInfo method = GetMethodFromStackTrace(_stackTrace!);
+
+            // The _stackTrace can be either sbyte[] or Object[]. In the latter case,
+            // the first entry is the actual stack trace sbyte[], the rest are pointers
+            // to the method info objects. We only care about the first entry here.
+            sbyte[] stacktrace;
+            if (_stackTrace is object[] objArray)
+            {
+                stacktrace = (sbyte[])objArray[0];
+            }
+            else
+            {
+                Debug.Assert(_stackTrace is sbyte[], "_stackTrace should be sbyte[]");
+                stacktrace = (sbyte[])_stackTrace!;
+            }
+
+            if (stacktrace.Length == 0)
```

Comment:
Stack trace should never have a zero length. It is either null or it contains at least one frame.

Reply:
Okay. I can assert this instead. It was in the original FCall implementation.